pilot/pkg/networking/core/v1alpha3/httproute_test.go

```diff
@@ -438,18 +438,17 @@ func TestSidecarOutboundHTTPRouteConfigWithDuplicateHosts(t *testing.T) {
 			}},
 			map[string][]string{
 				"allow_any": {"*"},
-				"test-duplicate-domains.default.svc.cluster.local:80": {
+				"test-duplicate-domains.default:80": {
+					"test-duplicate-domains.default",
 					"test-duplicate-domains.default.svc.cluster.local",
 					"test-duplicate-domains",
 					"test-duplicate-domains.default.svc",
 				},
-				"test-duplicate-domains.default:80": {"test-duplicate-domains.default"},
 			},
 			map[string]string{
 				"allow_any": "PassthroughCluster",
 				// Virtual service destination takes precedence
-				"test-duplicate-domains.default.svc.cluster.local:80": "outbound|80||test-duplicate-domains.default",
-				"test-duplicate-domains.default:80":                   "outbound|80||test-duplicate-domains.default",
+				"test-duplicate-domains.default:80": "outbound|80||test-duplicate-domains.default",
```

Comment:
This change does not seem right

Reply:
In this case, merging two virtualhosts **test-duplicate-domains.default.svc.cluster.local:80** and **test-duplicate-domains.default:80** into single virtualhost **test-duplicate-domains.default:80**, the map key is virtualhost name, value is virtualhost destination.